larray/core.py

```diff
@@ -4078,25 +3679,146 @@ def aslarray(a):
+              With this option, the result will broadcast correctly
+              against the input array.
+        """,
+    'interpolation':
+        """
+            * interpolation : {'linear', 'lower', 'higher', 'midpoint', 'nearest'}
+
+              This optional parameter specifies the interpolation method to
+              use when the desired quantile lies between two data points ``i < j``:
+                * linear: ``i + (j - i) * fraction``, where ``fraction`` is the
+                  fractional part of the index surrounded by ``i`` and ``j``.
+                * lower: ``i``.
+                * higher: ``j``.
+                * nearest: ``i`` or ``j``, whichever is nearest.
+                * midpoint: ``(i + j) / 2``.
+        """
+}
+
+
+def _doc_agg_method(desc, by=False, action="perform",
+                    extra_args='', kwargs=''):
+    extra_args = extra_args.split(',') if extra_args else []
+    kwargs = kwargs.split(',') if kwargs else []
+
+    if by:
+        str_doc = "The {0} is {1}ed along all axes except the given one(s). " \
+                  "The default (no axis or group) is to {1} the {0} " \
+                  "over all the dimensions of the input array.".format(desc, action)
+    else:
+        str_doc = "Axis(es) or group(s) along the {0} is {1}ed. " \
+                  " The default (no axis or group) is to {1} the {0} " \
+                  "over all the dimensions of the input array.".format(desc, action)
+
+    args_doc = \
+        """
+        Parameters
+        ----------
+        {0}
+        \*args : None or int or str or Axis or Group or any combination of those
+
+            {1}
+
+            An axis can be referred by:
+
+            * its position (integer). Position can be a negative integer,
+              in which case it counts from the last to the first axis.
+            * its name (str or AxisReference). You can use either a simple
+              string ('axis_name') or the special variable x (x.axis_name).
+            * a variable (Axis). If the axis has been defined previously
+              and assigned to a variable, you can pass it as argument.
+
+            You may not want to {2} the {3} over a whole axis but
+            over a selection of specific labels. To do so, you have several
+            possibilities:
+
+            * ('a1, a3, a5') : labels separated by commas in a string
+            * (a='a1, a3, a5') : in case of possible ambiguity, i.e. if labels
+              can belong to more than one axis, you must precise the axis.
+            * ('a1:a5:2') : select labels using a slice (syntax is 'start:end:step')
+            * ('a1:a3; a10:a12') : create several groups with semicolons.
+              Names are simply given by the concatenation of labels
+              (here: 'a1,a2,a3' and 'a10,a11,a12')
+            * ('a1:a3 >> a1_3; a10:a12 >> a12_12') : operator ' >> '
+              allows to rename groups.
```

Comment:
I think this section about groups would be clearer for the user if it was explained in steps:
* coma list or slice
* ambiguous labels need their axis to be specified using kwarg OR 'axis_name[]' OR x.axis_name[]
* several groups are possible
* any group can be renamed

The main problem with this explanation is that it is easy to miss the fact you can do all the above without using the string syntax, which I prefer users to avoid (in their models at least).